test/uk/gov/hmrc/helptosavestub/controllers/SquidControllerSpec.scala

```diff
@@ -0,0 +1,983 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestub.controllers
+
+import play.api.test._
+import org.scalatest.mock.MockitoSugar
+import play.api.i18n.MessagesApi
+import play.api.test.Helpers._
+import play.api.libs.json.{JsDefined, JsValue, Json}
+import play.api.libs.streams.Accumulator
+import play.api.mvc.{AnyContentAsEmpty, Result}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+import uk.gov.hmrc.helptosavestub.Constants._
+
+import scala.concurrent.Future
+
+class SquidControllerSpec extends UnitSpec with WithFakeApplication with MockitoSugar {
+  private val injector = fakeApplication.injector
+
+  private def messagesApi = injector.instanceOf[MessagesApi]
+
+  private def noCAKeyMap = Map[String, Map[String, String]]("wibble" -> Map())
+
+  private def noCAKeyJson = Json.toJson(noCAKeyMap)
```

Comment:
why are you using a map? Using a case class you can guarantee type safety at compile time